<!--
    Copyright 2018-2021 VMware, Inc.
    SPDX-License-Identifier: LGPL-3.0-or-later
-->

<p>The <code>ServletResponse.getOutputStream()</code> method returns a <code>ServletOutputStream</code> suitable for writing binary data in the response. However, in most usesthe response data is text and not binary data. For text, developers should use <code>ServletResponse.getWriter()</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class ServletResponseGetOutputStreamCheck {

  ServletOutputStream methodA1(ServletResponse servletResponse) {
    return servletResponse.getOutputStream();  //Noncompliant
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
ServletResponse.getWriter()
</pre>

